src/main/java/com/paskar/email/application/repositiory/EmailRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.paskar.email.application.repositiory;
+
+
+import com.paskar.email.application.console.Email;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+public interface EmailRepository {
+   void save(List<Email> email) throws IOException;
+    Email createNewEmail() throws IOException;
+    List<Email> findAll() throws IOException;
+    Optional<Email> findEmailsNearDeliveryDate(LocalDateTime currentDate) throws IOException;
```

Comment:
technically there might be several emails that are near to this date. So let's return List instead of optional